### PR TITLE
Fix missing integrity hash on preload

### PR DESCRIPTION
**Pull request by arnaud-ritti**
Has mentioned in issue #101 , integrity hash is missing into the link headers.

### CODE REVIEW

File: src/Asset/TagRenderer.php [hunk 1]
```diff
@@ -91,7 +91,7 @@ public function renderWebpackScriptTags(string $entryName, string $packageName =
                 $this->convertArrayToAttributes($attributes)
             );
 
-            $this->renderedFiles['scripts'][] = $attributes['src'];
+            $this->renderedFiles['scripts'][] = $attributes;
```
Comment (weaverryan) [inline src/Asset/TagRenderer.php]: Unfortunately, this is a BC break, as it changes the structure of what's returned from `getRenderedScripts()` and `getRenderedStyles()`.

Probably we need to add and maintain a second property - e.g. `$this->renderedFilesWithAttributes()`. Then, in `getRenderedScripts()` (and also for styles), we could add a new `bool $includeAttributes = false` argument. If that's true, then we return the `renderedFilesWithAttributes`.

**Review by weaverryan — COMMENTED** [review]
Sorry for the slow review! Thanks for this - there is just one minor backwards-compatibility concern we need to address!

File: src/Asset/TagRenderer.php (hunk 1 above)
Comment (arnaud-ritti) [reply src/Asset/TagRenderer.php]: Hi @weaverryan , Sorry for the delay. Thanks for your comment.
I fixed the BC break according your recommendations, if you want to check again.

Comment (ToshY) [reply src/Asset/TagRenderer.php]: @weaverryan @jrushlow This PR has been open for almost 2 years now, could either one spare some time to review it further and merge it? Thanks in advance.

**Comment by mustanggb** [comment]
Needs a reroll.

**Comment by ToshY** [comment]
@weaverryan @jrushlow  please take a look at this.

**Comment by ToshY** [comment]
@weaverryan @jrushlow please take a look at this.

**Comment by ToshY** [comment]
@weaverryan @jrushlow please take a look at this.

File: src/Asset/TagRenderer.php [hunk 2]
```diff
@@ -91,13 +92,14 @@ public function renderWebpackScriptTags(string $entryName, string $packageName =
                 $this->convertArrayToAttributes($attributes)
             );
 
-            $this->renderedFiles['scripts'][] = $attributes['src'];
+            $this->renderedFiles['scripts'][] = $attributes["src"];
```
Comment (jrushlow) [inline src/Asset/TagRenderer.php:95]:
```suggestion
            $this->renderedFiles['scripts'][] = $attributes['src'];
```

File: src/Asset/TagRenderer.php [hunk 3]
```diff
@@ -129,7 +131,8 @@ public function renderWebpackLinkTags(string $entryName, string $packageName = n
                 $this->convertArrayToAttributes($attributes)
             );
 
-            $this->renderedFiles['styles'][] = $attributes['href'];
+            $this->renderedFiles['styles'][] = $attributes["href"];
```
Comment (jrushlow) [inline src/Asset/TagRenderer.php:134]:
```suggestion
            $this->renderedFiles['styles'][] = $attributes['href'];
```

**Comment by ping-localhost** [comment]
Can you process the feedback @arnaud-ritti or would you like someone else to pick this up?

**Comment by ToshY** [comment]
Multiple other users have already reported this bug (in #101 , #210 , #220, #225) and others also offered to fix it, but the lack of communication and the delayed response times from the core maintainers is shockingly bad for Symfony standards. And while I fully understand that this is not something Ryan should have to worry about (regarding his health), it does not take away that the original issue #101 was created on `December 10, 2020` (and this PR roughly a year later on `January 12, 2022`). 

We are now nearing the end of 2024 and I have a feeling that this will not get merged for another 4 years at this rate.

Is there not a(nother) core maintainer that can take responsibility for this bundle and make sure that atleast bugfixes still have a chance of getting merged?

**Comment by Kocal** [comment]
Hi @ToshY, thanks for pointing this out.

Yes, things have slowed down a lot if not almost stopped, for the reasons you mentioned, but also for the time invested on the Symfony AssetMapper component. Note, this has also affected [Webpack Encore](https://github.com/symfony/webpack-encore). 

> but the lack of communication and the delayed response times from the core maintainers is shockingly bad for Symfony standards.

I know it can be frustrating, but maybe that Webpack isn't their priority, and above all, whether it's Symfony or not, at the end of the day it's still open-source and nobody owes you anything.

That said, I've been able to merge and create releases for a few weeks now. I'll do my best to handle the existing pull-requests, including this one!

Cheers!

**Comment by ToshY** [comment]
Thank you for your response @Kocal . 

> Yes, things have slowed down a lot if not almost stopped, for the reasons you mentioned, but also for the time invested on the Symfony AssetMapper component. Note, this has also affected [Webpack Encore](https://github.com/symfony/webpack-encore).

I'm somewhat (positively) surprised that Webpack Encore (in this case you) has made some updates in the last couple of days. That's encouraging to see.

> I know it can be frustrating, but maybe that Webpack isn't their priority, and above all, whether it's Symfony or not, at the end of the day it's still open-source and nobody owes you anything.

I am aware of that fact. It's just disappointing that this issue could have been resolved years ago if there was better communication and a more pro-active approach in merging open PRs.

**Comment by Kocal** [comment]
Unfortunately the checkbox "Allow edits and access to secrets by maintainers" is not checked, so I won't be able to push on this PR. 
I'm gonna open a new PR (with @arnaud-ritti's commits), fix some things, and merge ASAP.

**Comment by Kocal** [comment]
Released in https://github.com/symfony/webpack-encore-bundle/releases/tag/v2.2.0

**Pull request closed without merging**